app/controllers/katello/api/v2/host_subscriptions_controller.rb

```diff
@@ -43,6 +44,47 @@ def destroy
       respond_for_destroy(:resource => @host)
     end
 
+    api :POST, "/hosts/subscriptions/", N_("Register a host with subscription and information.")
+    param :name, String, :desc => N_("Name of the host"), :required => true
+    param :uuid, String, :desc => N_("UUID to use for registered host, random uuid is generated if not provided")
+    param :facts, Hash, :desc => N_("Key-value hash of subscription-manager facts, nesting uses a period delimiter (.)")
+    param :hypervisor_guest_uuids, Array, :desc => N_("UUIDs of the virtual guests from the host's hypervisor")
+    param :installed_products, Array, :desc => N_("List of products installed on the host") do
+      param :product_id, String, :desc => N_("Product id as listed from a host's installed products")
```

Comment:
Maybe we say "not the product ID field as seen when listing products from the products API" I realize that is a mouthful but it might help to clarify those reading the docs.